File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        hold = -prices[0]
        sell = 0
        
        for i in range(1, len(prices)):
            hold_old = hold  # Save the old value
            hold = max(hold, -prices[i])
            sell = max(sell, hold_old + prices[i])  # Use the old hold
        
        return sell

File: Data Structures & Algorithms/n-th-tribonacci-number/submission-0.java
class Solution {
    public int tribonacci(int n) {
        if(n==1)
        {
            return n;
        }
        
        int n1 = 0,n2 = 1,n3 = 1;
        for(int i = 1; i<=n;i++)
        {
            int n4 = n1+n2+n3;
            n1=n2;
            n2=n3;
            n3=n4;
        }

        return n1;
    }
}